Amp 276 refactor galaxy import

yingfeng-iu opened the pull request.

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/controller/FileUploadController.java:
```diff
@@ -147,49 +136,47 @@ public PrimaryfileSupplement handlePrimaryfileSupplementUpload(@PathVariable("id
-	}
-
-
+//    @PostMapping("/datalib/import/{id}")
```
@yingfeng-iu  can we remove commented code?

yingfeng-iu replied:
yes, defintely. i kept them in case there will be changes reverted and such. will remove before merge

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/controller/WorkflowController.java:
```diff
@@ -26,8 +28,19 @@
-	public GalaxyWorkflow[] getWorkflows() {		
-		return galaxyApiService.getWorkflows();
+	public List<Workflow> getWorkflows() {		
+		List<Workflow> workflows = galaxyApiService.getInstance().getWorkflowsClient().getWorkflows();
```
@yingfeng-iu can we refactor this and move all galaxy api code and move them to amp services rather than controllers? What do you both think? @vinita84

yingfeng-iu replied:
we do need a workflow controller on the amp side to take amp user actions. and retrieve galaxy workflow on the amp side is one such action. in general we shall not put complicated business logic in controller but in service.

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/controller/WorkflowController.java:
```diff
@@ -26,8 +28,19 @@
+//	 * Retrieve all workflows from Galaxy through its REST API.
+//	 * @return
+//	 */
+//	@GetMapping("/workflows")
```
Remove commented code, that is not needed, we can always go back in git history to get the code again, if needed.

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/model/galaxy/GalaxyApiKey.java:
```diff
@@ -3,6 +3,8 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+// TODO This class is not used anymore and can probably be removed.
```
Can we remove it, if it's not needed?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/model/galaxy/GalaxyWorkflow.java:
```diff
@@ -4,6 +4,8 @@
 import lombok.NoArgsConstructor;
 import lombok.NonNull;
 
+// TODO This class is not used anymore and can probably be removed.
```
Can we get rid of this as well? if not needed?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/service/impl/FileStorageServiceImpl.java:
```diff
@@ -209,4 +221,42 @@ else if (supplement instanceof PrimaryfileSupplement) {
+	/**
+	 * @see edu.indiana.dlib.amppd.service.FileStorageService.uploadFileToGalaxy(String,String)
+	 */
+	public ClientResponse uploadFileToGalaxy(String filePath, String libraryName) {
```
Would it make more sense to return boolean or custom response, instead of `ClientResponse` so its easier to implement?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/service/impl/FileStorageServiceImpl.java:
```diff
@@ -209,4 +221,42 @@ else if (supplement instanceof PrimaryfileSupplement) {
+	public ClientResponse uploadFileToGalaxy(String filePath, String libraryName) {
+		GalaxyInstance galaxyInstance = galaxyApiService.getInstance();
+		final LibrariesClient libraryClient = galaxyInstance.getLibrariesClient();
+		String msg = "Uploading file from Amppd file system to Galaxy data library... File path: " + filePath + "\t Galaxy Instance: " + galaxyInstance.getGalaxyUrl() + "\t Galaxy Library:" + libraryName;
```
Should we implement `i18n` for these messages? so its easier to change later on?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/service/impl/GalaxyApiServiceImpl.java:
```diff
@@ -34,39 +32,77 @@
-		GalaxyWorkflow[] workflows = restTemplate.getForObject(url, GalaxyWorkflow[].class);
-		log.info("Current workflows in Galaxy: " + workflows);
-		return workflows;
+	public GalaxyInstance getInstance() {
```
@yingfeng-iu can we implement exception handling, incase galaxy returns an error?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/service/impl/GalaxyApiServiceImpl.java:
```diff
@@ -34,39 +32,77 @@
+		user.setApiKey(key);
+		return key;
 
+//		/* Galaxy api key can be obtained via a GET request with baseauth, for ex.
```
Can we remove commented code?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/web/GalaxyRestTemplateFactory.java:
```diff
@@ -11,6 +11,8 @@
 import lombok.Getter;
 import lombok.Setter;
 
+// TODO This class is not used anymore and can probably be removed.
```
Same here, if not needed, maybe we can remove it from commit?

adeelahmad commented on src/main/java/edu/indiana/dlib/amppd/web/HttpComponentsClientHttpRequestFactoryBasicAuth.java:
```diff
@@ -12,6 +12,8 @@
 import org.springframework.http.HttpMethod;
 import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
 
+// TODO This class is not used anymore and can probably be removed.
```
Same as others.

yingfeng-iu replied:
we might want to keep this one in case we impl anything for AMP rest api using basic auth. i think it's ok to decide this later.

adeelahmad commented on src/test/java/edu/indiana/dlib/amppd/controller/WorkflowControllerTests.java:
```diff
@@ -22,14 +21,15 @@
     			// TODO also need to add boolean expression for the case when no workflow exists
-//				jsonPath("$", hasSize(1))).andExpect(	
-						jsonPath("$[0].model_class").value("StoredWorkflow"));
+    			//jsonPath("$", hasSize(1))).andExpect(	
```
I am getting, we will redo this, once Galaxy for testing is in place? also, we might have to create some workflows using api before testing.

yingfeng-iu replied:
is there a way to bootstrap galaxy with some sample data? one workflow could be part of that sample data. a lot of unit tests touch galaxy will probably need such data. it would be best if there're some factory method to do this.

yingfeng-iu commented:
Adeel, I addressed all the comments you made. let me know if any other changes needed. I need to merge as AMP-279 relies on these changes. I can make further changes in AMP-279 if u request.
